azure_functions_worker/bindings/meta.py

```diff
@@ -3,18 +3,46 @@
 import sys
 import typing
 
-from .. import protos
+from azure_functions_worker.constants import HTTP, HTTP_TRIGGER
 
+from .. import protos
 from . import datumdef
 from . import generic
 from .shared_memory_data_transfer import SharedMemoryManager
+from ..http_v2 import HttpV2Registry
 
 PB_TYPE = 'rpc_data'
 PB_TYPE_DATA = 'data'
 PB_TYPE_RPC_SHARED_MEMORY = 'rpc_shared_memory'
 BINDING_REGISTRY = None
 
 
+def _check_http_input_type_annotation(bind_name: str, pytype: type) -> bool:
+    if HttpV2Registry.http_v2_enabled():
+        return HttpV2Registry.ext_base().RequestTrackerMeta \
+            .check_type(pytype)
+
+    binding = get_binding(bind_name)
+    return binding.check_input_type_annotation(pytype)
+
+
+def _check_http_output_type_annotation(bind_name: str, pytype: type) -> bool:
+    if HttpV2Registry.http_v2_enabled():
+        return HttpV2Registry.ext_base().ResponseTrackerMeta.check_type(pytype)
+
+    binding = get_binding(bind_name)
+    return binding.check_output_type_annotation(pytype)
+
+
+INPUT_TYPE_CHECK_OVERRIDE_MAP = {
+    HTTP_TRIGGER: _check_http_input_type_annotation
+}
+
+OUTPUT_TYPE_CHECK_OVERRIDE_MAP = {
```

Comment:
nit: move the maps to the top of the file with the other constants

Reply:
![image](https://github.com/Azure/azure-functions-python-worker/assets/12449837/63a923f0-87e9-49dc-81a9-21791fdedbb7)
have to keep them below function definitions.